go-controller/pkg/ovn/base_network_controller_secondary.go

```diff
@@ -0,0 +1,249 @@
+	case factory.PodType:
+		pod, ok := obj.(*kapi.Pod)
+		if !ok {
+			return fmt.Errorf("could not cast %T object to *knet.Pod", obj)
+		}
+		return bnc.ensurePod4SecondaryNetwork(pod, true)
+
+	default:
+		return fmt.Errorf("object type %s not supported", objType)
+	}
+}
+
+func (bnc *BaseNetworkController) UpdateSecondaryNetworkResourceCommon(objType reflect.Type, oldObj, newObj interface{}, inRetryCache bool) error {
+	switch objType {
+	case factory.PodType:
+		oldPod := oldObj.(*kapi.Pod)
+		newPod := newObj.(*kapi.Pod)
+
+		return bnc.ensurePod4SecondaryNetwork(newPod, inRetryCache || util.PodScheduled(oldPod) != util.PodScheduled(newPod))
+
+	default:
+		return fmt.Errorf("object type %s not supported", objType)
+	}
+}
+
+func (bnc *BaseNetworkController) DeleteSecondaryNetworkResourceCommon(objType reflect.Type, obj, cachedObj interface{}) error {
+	switch objType {
+	case factory.PodType:
+		var portInfo *lpInfo
+		pod := obj.(*kapi.Pod)
+
+		if cachedObj != nil {
+			portInfo = cachedObj.(*lpInfo)
+		}
+		return bnc.removePod4SecondaryNetwork(pod, portInfo)
+
+	default:
+		return fmt.Errorf("object type %s not supported", objType)
+	}
+}
+
+// ensurePod4SecondaryNetwork tries to set up a pod for secondary network. It returns nil on success and error
+// on failure; failure indicates the pod set up should be retried later.
+func (bnc *BaseNetworkController) ensurePod4SecondaryNetwork(pod *kapi.Pod, addPort bool) error {
+
+	// Try unscheduled pods later
+	if !util.PodScheduled(pod) {
+		return nil
+	}
+
+	if !util.PodWantsNetwork(pod) && !addPort {
+		return nil
+	}
+
+	// If a node does node have an assigned hostsubnet don't wait for the logical switch to appear
+	switchName, err := bnc.getExpectedSwitchName(pod)
+	if err != nil {
+		return err
+	}
+
+	if bnc.lsManager.IsNonHostSubnetSwitch(switchName) {
+		return nil
+	}
+
+	on, network, err := util.IsNetworkOnPod(pod, bnc.NetInfo)
+	if err != nil {
+		// configuration error, no need to retry, do not return error
+		klog.Errorf("Error getting network-attachment for pod %s/%s network %s: %v",
+			pod.Namespace, pod.Name, bnc.GetNetworkName(), err)
+		return nil
+	}
+
+	if !on {
+		// the pod is not attached to this specific network
+		klog.V(5).Infof("Pod %s/%s is not attached on this network controller %s error (%v) ",
+			pod.Namespace, pod.Name, bnc.GetNetworkName(), err)
+		return nil
+	}
+
+	nadName := util.GetNadName(network.Namespace, network.Name)
+
+	var libovsdbExecuteTime time.Duration
+
+	// Keep track of how long syncs take.
+	start := time.Now()
+	defer func() {
+		klog.Infof("[%s/%s] addLogicalPort for nad %s took %v, libovsdb time %v",
+			pod.Namespace, pod.Name, nadName, time.Since(start), libovsdbExecuteTime)
+	}()
+
+	ops, lsp, podAnnotation, newlyCreated, err := bnc.addLogicalPortToNetwork(pod, nadName, network)
+	if err != nil {
+		return err
+	}
+
+	recordOps, txOkCallBack, _, err := metrics.GetConfigDurationRecorder().AddOVN(bnc.nbClient, "pod", pod.Namespace,
```

Comment:
Does not make a look of sense to call `AddOVN` if we are not calling the corresponding `Start` and `End` methods.

Reply:
I agree. My thinking is what while I am not sure in the future we'd need it for secondary network and would like to keep a clue for the future. It is an no-op for secondary networks now. If you prefer to remove it, let me know...